src/TLuaInterpreter.cpp

```diff
@@ -10739,6 +10750,505 @@ int TLuaInterpreter::restartIrc(lua_State* L)
+    speechUnit->say(speechCurrent);
+    
+    return;
+}
+
+
+/** ttsQueueSpeech( input, opt. index )
+ *  Adds a line of text to the queue at the specified index.
+ *  If no index is provided, it will add it to the end of the queue.
+ */
+int TLuaInterpreter::ttsQueueSpeech(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    if(!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "ttsQueueText: bad argument #1 type (input as string expected, got %s!)", luaL_typename(L, 1));
+        lua_error(L);
+        return 1;
+    }
+    
+    QString inputText = lua_tostring(L, 1);
+    int index;
+    
+    if(lua_gettop(L) > 1) {
+        if(!lua_isnumber(L, 2)) {
+            lua_pushfstring(L, "ttsQueueText: bad argument #2 type (optional index as number expected, got %s!)", luaL_typename(L, 1));
+            lua_error(L);
+            return 1;
+        }
+        
+        index = lua_tonumber(L, 2);
+        index--;
+        
+        if(index < 0) {
+            index = 0;
+		}
+        
+		if(index > speechQueue.size()) {
+            index = speechQueue.size();
+		}
+    }
+    else {
+        index = speechQueue.size();
+    }
+
+    speechQueue.insert(index, inputText);
+    
+    TEvent event;
+    Host& host = getHostFromLua(L);
+    event.mArgumentList.append(QLatin1String("ttsSpeechQueued"));
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append(inputText);
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append(QString::number(index));
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    host.raiseEvent(event);
+    
+    TLuaInterpreter::ttsStateChanged(speechUnit->state());
+    
+    return 0;
+}
+
+/** ttsGetSpeechQueue( opt. index )
+ *  Returns the queued text at index.
+ *  If no index exists, returns false.
+ *  If no index provided, returns a table with the entire queue.
+ */
+int TLuaInterpreter::ttsGetSpeechQueue(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild();
+    
+    if(lua_gettop(L) > 0) {
+        if(!lua_isnumber(L, 1)) {
+            lua_pushfstring(L, "ttsGetspeechQueue. bad argument #1 type (optional index as number expected, got %s!)", luaL_typename(L, 1));
+            lua_error(L);
+            return 1;
+        }
+        
+        int index = lua_tonumber(L, 1);
+        index--;
+        
+        if(index < 0 || index > speechQueue.size()) {
+            lua_pushboolean(L, false);
+            return 1;
+        }
+        
+        lua_pushstring(L, speechQueue.at(index).toLatin1().constData());
+        return 1;
+    }
+    
+    lua_newtable(L);
+    
+    for(int i=0;i<speechQueue.size();i++) {
+        lua_pushnumber(L, i + 1);
+        lua_pushstring(L, speechQueue.at(i).toUtf8().constData());
+        lua_settable(L, -3);
+    }
+    
+    return 1;
+}
+
+/** ttsPauseSpeech( opt. pause )
+ *  Pauses speech if passed true, resumes if passed false.
+ *  If no input, returns whether or not the queue is paused.
```

Comment:
Another lying comment. The function doesn't return anything. And I wouldn't return the current paused state either as I'd want an "IsSpeechPaused" function for that.

In truth, I'd probably use a "pauseSpeech" "resumeSpeech" function similar to "enableTrigger" and "disableTrigger"